pkg/reporter/reporter_mock.go

```diff
@@ -0,0 +1,60 @@
+// Copyright 2024 The Authors (see AUTHORS file)
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package reporter
```

Comment:
What is this for?

Reply:
It will be used when testing the plan/apply commands. I am trying to send smaller PRs and i dont like git-town.

Reply:
Does it need to be package-public?

Reply:
Yes, will be used in other packages.